src/server/scripts/Outland/zone_shadowmoon_valley.cpp

```diff
@@ -571,64 +589,112 @@ class npc_dragonmaw_peon : public CreatureScript
     {
         npc_dragonmaw_peonAI(Creature* creature) : ScriptedAI(creature) { }
 
+        EventMap events;
+        EventMap events2;
         uint64 PlayerGUID;
         bool Tapped;
-        uint32 PoisonTimer;
 
         void Reset()
         {
+            events.Reset();
+            events2.Reset();
             PlayerGUID = 0;
             Tapped = false;
-            PoisonTimer = 0;
+        }
+
+        void EnterCombat(Unit* /*who*/)
+        {
+            events.ScheduleEvent(EVENT_KICK, urand(5000, 10000));
+            events.ScheduleEvent(EVENT_SUNDER, urand(5000, 10000));
         }
 
         void SpellHit(Unit* caster, const SpellInfo* spell)
         {
             if (!caster)
                 return;
 
-            if (caster->GetTypeId() == TYPEID_PLAYER && spell->Id == 40468 && !Tapped)
-            {
-                PlayerGUID = caster->GetGUID();
+            PlayerGUID = caster->GetGUID();
 
+            if (caster->GetTypeId() == TYPEID_PLAYER && spell->Id == SPELL_POISON && !Tapped)
+            {
                 Tapped = true;
-                float x, y, z;
                 caster->GetClosePoint(x, y, z, me->GetObjectSize());
-
-                me->SetWalk(false);
-                me->GetMotionMaster()->MovePoint(1, x, y, z);
+                Talk(SAY_1);
+                events2.ScheduleEvent(EVENT_WALK_TO_MUTTON, 0);
             }
         }
 
-        void MovementInform(uint32 type, uint32 id)
+        void MovementInform(uint32 /*type*/, uint32 id)
         {
-            if (type != POINT_MOTION_TYPE)
-                return;
+            if (id == 1)
+            {
+                if (GameObject* food = me->FindNearestGameObject(185893, 5.0f))
+                    me->SetFacingToObject(food);
+                me->HandleEmoteCommand(EMOTE_ONESHOT_EAT);
+                events2.ScheduleEvent(EVENT_POISONED, 5000);
+            }
+        }
 
-            if (id)
+        void CreditPlayer()
+        {
+            if (PlayerGUID)
             {
-                me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_EAT);
-                PoisonTimer = 15000;
+                Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID);
+                if (player && player->GetQuestStatus(11020) == QUEST_STATUS_INCOMPLETE)
+                    player->KilledMonsterCredit(23209, 0);
             }
         }
 
         void UpdateAI(uint32 diff)
         {
-            if (PoisonTimer)
+            events.Update(diff);
```

Comment:
Just for notice, you can have the same eventmap for both Out and in combat. As long as the checks out of combat are inside !updatevictim() they will never be triggered. So you can get rid of 1 of the events variable.